### PR TITLE
Init CMakeLists

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+
+find_package(GTest REQUIRED)
+
+include_directories(${GMOCK_ROOT}/include)
+include_directories(${GMOCK_ROOT}/gtest/include)
+include_directories(${GTEST_INCLUDE_DIRS})
+
+include_directories(./src/Header)
+
```
Comment (LinRenHong) [inline CMakeLists.txt:21]: Please add comment

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#will be used when other branch is merged
+#add_executable(TestRequestParser 
+#		src/Implement/UnitTestRequestParser.cpp
+#		src/Implement/RequestParser.cpp
+#		src/Header/RequestParser.h
+#		src/Header/OperateData.h
+#		)
+
+#target_link_libraries(TestRequestParser pthread GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main)
```
Comment (harrisonyei) [inline CMakeLists.txt]: Please add newline at the end of file.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+set (CMAKE_CXX_STANDARD 14)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+
+find_package(GTest REQUIRED)
+
+include_directories(${GMOCK_ROOT}/include)
+include_directories(${GMOCK_ROOT}/gtest/include)
+include_directories(${GTEST_INCLUDE_DIRS})
+
+include_directories(./src/Header)
```
Comment (harrisonyei) [inline CMakeLists.txt]: Avoid using relative path, 
use CMAKE_SOURCE_DIR, or PROJECT_SOURCE_DIR, or CMAKE_CURRENT_SOURCE_DIR instead.

**Review by harrisonyei — COMMENTED** [review]
Viewed, added some personal suggestions.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+cmake_minimum_required(VERSION 3.20)
+project(TestRequestParser)
+enable_testing()
+
+set (CMAKE_CXX_STANDARD 14)
```
Comment (Geas77) [inline CMakeLists.txt]: Alignment Problem

File: CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+
+find_package(GTest REQUIRED)
+
```
Comment (Geas77) [inline CMakeLists.txt]: I think need to add a if else to check find package or not like
if (GTEST_FOUND)
include_directories(${GTEST_INCLUDE_DIRS})
endif (GTEST_FOUND)

File: CMakeLists.txt (hunk 3 above)
Comment (john0312) [reply CMakeLists.txt]: This is a reasonable code review.

File: CMakeLists.txt (hunk 4 above)
Comment (john0312) [inline CMakeLists.txt]: Extra space.

File: CMakeLists.txt (hunk 2 above)
Comment (john0312) [reply CMakeLists.txt]: This is a reasonable review.

Comment (john0312) [inline CMakeLists.txt]: [Optional] Consider moving pthread to an if block that is restricted to *nix.

**Review by john0312 — APPROVED** [review]
Please fix reviews before merging.

File: CMakeLists.txt (hunk 4 above)
Comment (john0312) [reply CMakeLists.txt]: This is a reasonable review.

File: CMakeLists.txt (hunk 5 above)
Comment (john0312) [reply CMakeLists.txt]: GTest is "REQUIRED", so it'll fail if it's not found.

File: CMakeLists.txt (hunk 4 above)
Comment (younger15) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (younger15) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (younger15) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (younger15) [reply CMakeLists.txt]: Done

**Review by LinRenHong — APPROVED** [review]

**Review by will889889 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (john0312) [reply CMakeLists.txt:21]: You meant when to uncomment the below? If so, this is a reasonable review. Please add TODO on when to uncomment.

**Review by john0312 — APPROVED** [review]

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+#			src/Implement/RequestParser.cpp
+#			src/Header/RequestParser.h
+#			src/Implement/RequestHandler.cpp
+#			src/Header/RequestHandler.h
+#			src/Header/OperateData.h
+#			)
+#	target_link_libraries(TestRequestHandler pthread GTest::gtest GTest::gtest_main GTest::gmock GTest::gmock_main)
+#	gtest_discover_tests(TestRequestHandler)
+#	add_test(RequestHandler_test TestRequestHandler)
+endif(UNIX)
```
Comment (john0312) [inline CMakeLists.txt]: Consider adding else and cause it to fail.

Comment (younger15) [reply CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 1 above)
Comment (younger15) [reply CMakeLists.txt:21]: Done

**Pull request merged**